### ISSUE TITLE
[Harbor v2.11.0] error while putting blob to local repo, http status code: 500 (S3)

### ISSUE THREAD

**Issue by remkolems** [goharbor/harbor#20599]
I need assistance in solving the issue I encounter, Don't know if it is a bug/regression. It has something to do with the S3 driver API implementation. Searched throughout the web/this repo to no avail. I've used Harbor and this particular configuration since the beginning of this year without any issues.

All of a sudden and since `Harbor v2.11.0` I encounter the following error message in my `harbor-core` log:

```shell
2024-06-15T13:25:00Z [WARNING] [/server/middleware/repoproxy/proxy.go:208]: Artifact: hub.docker.com/library/redis:, digest:sha256:892b41c092a599f76c30b48e9dcfb185ce8cea3560970b1c4f2745c89bb34344 is not found in proxy cache, fetch it from remote repo
2024-06-15T13:25:00Z [WARNING] [/server/middleware/repoproxy/proxy.go:208]: Artifact: hub.docker.com/library/busybox:, digest:sha256:50aa4698fa6262977cff89181b2664b99d8a56dbca847bf62f2ef04854597cf8 is not found in proxy cache, fetch it from remote repo
2024-06-15T13:25:00Z [WARNING] [/server/middleware/repoproxy/proxy.go:208]: Artifact: hub.docker.com/searxng/searxng:, digest:sha256:a2de573895707a80139ca373eb7df14650538c1f4187fb4c5e0fe9fce8fccab5 is not found in proxy cache, fetch it from remote repo
2024-06-15T13:25:01Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:01Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:01Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:01Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:01Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:01Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:01Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:01Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:01Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:02Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:02Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:03Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:03Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:03Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:03Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:04Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:04Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
2024-06-15T13:25:04Z [ERROR] [/controller/proxy/controller.go:289]: error while putting blob to local repo, http status code: 500, body: {"errors":[{"code":"UNKNOWN","message":"unknown error","detail":{"DriverName":"s3aws","Enclosed":{"RequestFailure":{}}}}]}
```

I use:
- Minio S3 as backend storage on a separate node: MinIO VERSION 2024-06-13T22:53:53Z (current version)
- docker-compose online installer of Harbor with Trivy: Harbor Version v2.11.0-70255684 

What I did to solve issue:
- Completely uninstalled and reconfigured MinIO S3 from `RELEASE.2024-02-09T21-25-16Z` all the way up to `MinIO VERSION 2024-06-13T22:53:53Z)`
- Completely uninstalled Harbor and all of its directories throughout my `Ubuntu 22.04.4 LTS (GNU/Linux 6.5.0-35-generic x86_64)` system and reconfigured. Why is the `/data` directory volume used outside of the main Harbor directory? Why not all volumes confined under the Harbor directory?
- Tested MinIO throughout. Works as expected. Other software apps can reach and upload data to MinIO S3 (GitLab, Portainer) from this same node.

Consequences:
- The pulled container images are now stored locally, instead of the configured S3 bucket. The bucket and Harbor stays empty. Harbor UI `We couldn't find any repositories!`. 
- This issue in particular defeats the whole purpose now of Harbor and all of its useful functions, when using S3.
- Do note: Harbor without S3 storage_service works as expected again.

My `harbor.yml` config in relation to S3:
```yml
# Harbor Storage settings by default is using /data dir on local filesystem
# Uncomment storage_service setting If you want to using external storage
storage_service:
  #   # ca_bundle is the path to the custom root ca certificate, which will be injected into the truststore
  #   # of registry's containers.  This is usually needed when the user hosts a internal storage with self signed certificate.
  #   ca_bundle:

  #   # storage backend, default is filesystem, options include filesystem, azure, gcs, s3, swift and oss
  #   # for more info about this configuration please refer https://distribution.github.io/distribution/about/configuration/
  # filesystem:
  #     maxthreads: 100
  #   # set disable to true when you want to disable registry redirect
  redirect:
    disable: true
  #   # S3 backend config
  #     - example given https://distribution.github.io/distribution/about/configuration/#list-of-configuration-options
  #     - uncomment only if supported by respective backend
  s3:
    accesskey: <removed>
    secretkey: <removed>
    region: eu-west-01
    regionendpoint: https://s3.<removed>.com
    forcepathstyle: true
    #   accelerate: false
    bucket: harbor-registry
    encrypt: false
    #   keyid: mykeyid
    secure: true
    v4auth: true
    chunksize: 5242880
    #   multipartcopychunksize: 33554432
    #   multipartcopymaxconcurrency: 100
    #   multipartcopythresholdsize: 33554432
    rootdirectory: / # /s3/object/name/prefix
  #   usedualstack: false
  #   loglevel: debug
```

**Comment by zyyw** [comment]
Do you have a chance to go through this issue? The error messages are the same. Might be related:
* https://github.com/goharbor/harbor/issues/10327

**Comment by remkolems** [comment]
Thanks for your feedback!

I suspected something indeed with the chunksize initially. I have played around the last couple of days with all kind of settings That didn't solve it for me.

But I've managed to get Harbor running again just now.

Setup debug mode where I could on every major application and indeed system wide issues this morning. Not related in particular to any application whatsoever as it seems, but it appeared to be a very weird progressing failing routing issue. No clue how that happened. Still investigating in-depth.

In summary: this morning I've setup a completely new config of MinIO, Harbor and all major applications again. Configured all from scratch again (no backups used at all). In addition to that, I also changed the complete DNS routing (SSL termination with HAProxy) scheme to another internal https domain. 

The latter solved it. Thus it was a routing issue after all.
